File: src/test/java/uk/gov/justice/hmpps/offenderevents/services/wiremock/HMPPSAuthExtension.java
package uk.gov.justice.hmpps.offenderevents.services.wiremock;

import org.junit.jupiter.api.extension.AfterAllCallback;
import org.junit.jupiter.api.extension.BeforeAllCallback;
import org.junit.jupiter.api.extension.BeforeEachCallback;
import org.junit.jupiter.api.extension.ExtensionContext;

public class HMPPSAuthExtension implements BeforeAllCallback, AfterAllCallback, BeforeEachCallback {
    public static final HMPPSAuthMockServer server = new HMPPSAuthMockServer();

    @Override
    public void afterAll(ExtensionContext context) {
        server.stop();
    }

    @Override
    public void beforeAll(ExtensionContext context) {
        server.stubGrantToken();
        server.start();
    }

    @Override
    public void beforeEach(ExtensionContext context) {
        server.resetRequests();
    }
}

File: src/test/kotlin/uk/gov/justice/hmpps/offenderevents/services/wiremock/HMPPSAuthExtension.kt
package uk.gov.justice.hmpps.offenderevents.services.wiremock

import org.junit.jupiter.api.extension.AfterAllCallback
import org.junit.jupiter.api.extension.BeforeAllCallback
import org.junit.jupiter.api.extension.BeforeEachCallback
import org.junit.jupiter.api.extension.ExtensionContext

class HMPPSAuthExtension : BeforeAllCallback, AfterAllCallback, BeforeEachCallback {
  override fun afterAll(context: ExtensionContext): Unit = server.stop()

  override fun beforeAll(context: ExtensionContext) {
    server.stubGrantToken()
    server.start()
  }

  override fun beforeEach(context: ExtensionContext): Unit = server.resetRequests()

  companion object {
    val server = HMPPSAuthMockServer()
  }
}
